### PR TITLE
Add Component Search for License None/Any

### PR DESCRIPTION
**Pull request by davakos**
## Description
Primary Change: Add option to search components with No Licenses or Any Licenses
Other Updates:
- Update third party components, e.g., actions, nodejs 8.15 => 8.16, lingui v3 => v4, fix any breakages
- Reorder imports

## Motivation and Context
Adds ability to search for components with no associated license or any associated license.
Use latest supported versions of third party dependencies.

## How Has This Been Tested?
Add new unit tests for license [none, any] cases and ensure existing unit tests run.
No new mock data needed since existing component mocks already had components with/without licenses.
Manually check areas that may have been impacted by dependency updates, such as forms, look-and-feel, translations. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [X] My code follows conforms to the coding standards.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

## Pic
Embed something funny here

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -13,7 +13,6 @@ import store, { AppDispatch } from "app/store";
 import { Provider } from "react-redux";
 import { i18n } from "@lingui/core";
 import { I18nProvider } from "@lingui/react";
-import { en } from "make-plural/plurals";
```
Comment (davakos) [inline ui/src/App.tsx:16]: no longer required for lingui v4

File: ui/src/api/__tests__/client.test.ts [hunk 2]
```diff
@@ -1,14 +1,9 @@
-import axios, { AxiosRequestConfig } from "axios";
```
Comment (davakos) [inline ui/src/api/__tests__/client.test.ts:1]: use consistent import ordering across files

File: ui/src/api/__tests__/client.test.ts [hunk 3]
```diff
@@ -133,7 +133,7 @@ describe("api client", () => {
 		it("should reload current page", () => {
 			// mock window.location.reload
 			const globalWindow = global.window;
-			global.window = Object.create(window);
+			global.window ??= Object.create(window);
```
Comment (davakos) [inline ui/src/api/__tests__/client.test.ts:136]: workaround to suppress the error, `TypeError: Cannot redefine property: window` appearing after NodeJS 18.16 upgrade. this looks like it's a bug that has been fixed upstream in NodeJS 20 but hasn't been back-ported to v18 yet.

File: ui/src/api/server.ts [hunk 4]
```diff
@@ -826,9 +826,10 @@ export function makeServer() {
 				{
 					id: generateId(),
 					name: "admin api key",
-					created: DateTime.utc()
-						.minus({ days: Math.floor(Math.random() * 365) })
-						.toJSON(),
+					created:
+						DateTime.utc()
+							.minus({ days: Math.floor(Math.random() * 365) })
+							.toJSON() ?? undefined,
```
Comment (davakos) [inline ui/src/api/server.ts:832]: account for stricter luxon return types after upgrades

File: ui/src/components/DraggableDialog.tsx [hunk 5]
```diff
@@ -52,7 +52,7 @@ function PaperComponent(props: PaperProps) {
 	);
 }
 
-interface DraggableDialogProps extends DialogProps {
+interface DraggableDialogProps extends Omit<DialogProps, "content"> {
```
Comment (davakos) [inline ui/src/components/DraggableDialog.tsx:55]: MUI updates cause conflict between a `content` property of `DialogProps` and field added below for this custom interface, so omit `content` field from the inherited interface

File: ui/src/components/__tests__/DateTimeCell.test.tsx [hunk 6]
```diff
@@ -71,7 +71,9 @@ describe("ExpiringDateTimeCell component", () => {
 
 	it("future date displays no expiration warning", () => {
 		render(
-			<ExpiringDateTimeCell value={DateTime.utc().plus({ days: 1 }).toJSON()} />
+			<ExpiringDateTimeCell
```
Comment (davakos) [inline ui/src/components/__tests__/DateTimeCell.test.tsx:74]: account for stricter luxon return types

File: ui/src/pages/SearchPage.tsx [hunk 7]
```diff
@@ -421,6 +421,8 @@ type MatchDateT = "null" | "lt" | "exact" | "gt" | "bt";
 // components API does not include null matcher
 type MatchDateT = "lt" | "gt";
 
+type MatchLicenseT = "null" | "icontains" | "exact";
```
Comment (davakos) [inline ui/src/pages/SearchPage.tsx:424]: feature: changes to add support for component license none|any values start here

File: ui/src/pages/__tests__/SearchPage.test.tsx [hunk 8]
```diff
@@ -186,7 +186,7 @@ describe("SearchPage component", () => {
 					[
 						"License Match",
 						/license match/i,
-						["Contains", "Exact"],
+						["No License", "Any License", "Contains", "Exact"],
```
Comment (davakos) [inline ui/src/pages/__tests__/SearchPage.test.tsx:189]: updated tests to support component license none|any

File: ui/src/test-utils.tsx [hunk 9]
```diff
@@ -14,7 +14,6 @@ import userEvent from "@testing-library/user-event";
 import { Provider } from "react-redux";
 import { i18n } from "@lingui/core";
 import { I18nProvider } from "@lingui/react";
-import { en } from "make-plural/plurals";
```
Comment (davakos) [inline ui/src/test-utils.tsx:17]: no longer needed in lingui v4

**Review by davakos — COMMENTED** [review]
clarification of changes

File: ui/src/api/__tests__/client.test.ts (hunk 3 above)
Comment (davakos) [reply ui/src/api/__tests__/client.test.ts:136]: I wasn't planning to upgrade to NJS 20 until it's LTS ~ Nov 2023.
I was hoping that the bug fix for this would be back-ported to 18.x, but there hasn't been a release to 18.x since 18.16.0 on 12 Apr

**Review by g-marconet — APPROVED** [review]
LGTM, let's just make sure to document any tech debt so that we can circle back to it.
